File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        map<char,char> mpp1;
        map<char,char> mpp2;

        if(s.size() != t.size()) return false;
        for(int i = 0;i<s.size();i++)
        {
            if(mpp1.count(s[i]))
            {
                if(mpp1[s[i]] != t[i]) return false;
               
            }
             else
                mpp1[s[i]] = t[i];

            if(mpp2.count(t[i]))
            {
                if(mpp2[t[i]] != s[i]) return false;

            }   
            else
            mpp2[t[i]] = s[i];
           
        }
        
        return true;

        
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        
        HashMap<Character, Character> mpp1 = new HashMap<>();
        HashMap<Character, Character> mpp2;

        if (s.length() != t.length()) return false;

        mpp2 = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {

            if (mpp1.containsKey(s.charAt(i))) {
                if (mpp1.get(s.charAt(i)) != t.charAt(i))
                    return false;
            } else {
                mpp1.put(s.charAt(i), t.charAt(i));
            }

            if (mpp2.containsKey(t.charAt(i))) {
                if (mpp2.get(t.charAt(i)) != s.charAt(i))
                    return false;
            } else {
                mpp2.put(t.charAt(i), s.charAt(i));
            }
        }

        return true;
    }
}